### ISSUE TITLE
Documind core fails to install required libreoffice

### ISSUE THREAD

**Issue by cjfagerstrom** [DocumindHQ/documind#13]
When processing a PDF I get the error `Error converting .html to .pdf: Error: Could not find soffice binary`. Tracing through the code, Documind core checks for Libreoffice and installs it as a dependency in the `install-dependencies.js` file, but that doesn't happen when installing Documind on MacOS 15.1.

**Comment by Tammilore** [comment]
Hi @cjfagerstrom 

Thanks for reporting this issue! I'll investigate why the script isn't working properly on macOS 15.1 and will update this thread once I have a fix.

In the meantime, you can try manually installing [LibreOffice ](https://www.libreoffice.org/)on your system to see if that resolves the issue.

**Comment by cjfagerstrom** [comment]
@Tammilore I did manually install it, but I'm constantly getting hit with a 403 error from Axios saying `ERR_BAD_REQUEST`, so I haven't been able to get any further.

**Comment by Tammilore** [comment]
@cjfagerstrom Could you please share the full error log?

**Comment by cjfagerstrom** [comment]
@Tammilore [Documind Error Log.txt](https://github.com/user-attachments/files/17979254/Documind.Error.Log.txt)

**Comment by Tammilore** [comment]
@cjfagerstrom Can you check that your OpenAI account is at least on Tier 1 and has access to models like gpt-4o and gpt-4o-mini.

Then try generating a new API key and test with that. Also, don’t forget to disable your current key since it’s exposed in the logs. Let me know how it works out.

**Comment by cjfagerstrom** [comment]
@Tammilore It is on Tier 1 and does have access to both of those models.

I'll try generating a new key this evening and see if that works - that key was created prior to upgrading to Tier 1. Thanks for the reminder - I was going to scrub the logs of any keys, but didn't get a chance to look through them before posting and only had a couple dollars credit on the account anyways. I've revoked both keys so I can keep my $3 lol

**Comment by cjfagerstrom** [comment]
@Tammilore Still getting the same error. I've created a new key and after running documind for the first time it changed the status in OpenAI's dashboard from never used to today's date, but still getting 403. Permissions on the key are set for all.

**Comment by Tammilore** [comment]
Hey @cjfagerstrom , please share the logs.

Also, could you try making a direct API request on OpenAI using 4o and mini and let me know what the response is?

**Comment by cjfagerstrom** [comment]
@Tammilore I've been trying to troubleshoot this more and when I use Postman to send a POST request to `https://api.openai.com/v1/chat/completions`, it works.
```{
    "model": "gpt-4o-mini",
    "messages": [
        {
            "role": "system",
            "content": "\\n    Convert the following document page to markdown.\\n    Return only the markdown with no explanation text. Do not include deliminators like '''markdown.\\n    You must include all information on the page. Do not exclude headers, footers, or subtext.\\n",
            "images": ["iVBORw0KGgoAAAA....]}]}``` returns ```{
    "id": "chatcmpl-AqrcK8KkgHrToDU0iV01odt4tsA73",
    "object": "chat.completion",
    "created": 1737162524,
    "model": "gpt-4o-mini-2024-07-18",
    "choices": [
        {
            "index": 0,
            "message": {
                "role": "assistant",
                "content": "# Document Title\n\n## Section Header\n\nThis is a paragraph of text in the document. It contains important information about the topic being discussed. \n\n### Subsection Header\n\n- Bullet point one\n- Bullet point two\n- Bullet point three\n\n1. ....
```

But running my test script in Documind yields a 403:
```Error in OpenAI completion AxiosError: Request failed with status code 403
    at settle (/Users/Chris/Dev/documind/node_modules/axios/dist/node/axios.cjs:2019:12)
    at BrotliDecompress.handleStreamEnd (/Users/Chris/Dev/documind/node_modules/axios/dist/node/axios.cjs:3135:11)
    at BrotliDecompress.emit (node:events:529:35)
    at endReadableNT (node:internal/streams/readable:1368:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
    at Axios.request (/Users/Chris/Dev/documind/node_modules/axios/dist/node/axios.cjs:4287:41)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async getCompletion (/Users/Chris/Dev/documind/core/dist/openAI.js:53:26)
    at async processPage (/Users/Chris/Dev/documind/core/dist/index.js:113:64) {
  code: 'ERR_BAD_REQUEST',
  config: {
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    },
    adapter: [ 'xhr', 'http', 'fetch' ],
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    env: { FormData: [Function], Blob: [class Blob] },
    validateStatus: [Function: validateStatus],
    headers: Object [AxiosHeaders] {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': 'application/json',
      Authorization: 'Bearer REDACTED',
      'User-Agent': 'axios/1.7.7',
      'Content-Length': '341411',
      'Accept-Encoding': 'gzip, compress, deflate, br'
    },
    method: 'post',
    url: 'https://api.openai.com/v1/chat/completions',...
```

I've also verified that I can use axios to get the file from the public URL and console log out the file with a HTTP 200 response from axios.

**Comment by cjfagerstrom** [comment]
@Tammilore It would appear something in your update from 4 days ago resolved my issues as I am no longer getting a 403. I did re-download and run `npm install documind` last night, but I moved my code into a backup folder and my IDE's terminal didn't indicate I was working in a different directory after moving the code, but I think it was as today I closed and reopened my IDE and the terminal updated to show the backup directory. When I moved back into the new directory and ran `npm install documind` again, this time it worked and it was now showing the markdown console log that you added.
